libethashseal/Ethash.h

```diff
@@ -69,7 +69,13 @@ class Ethash: public SealEngineBase
     static h256 seedHash(BlockHeader const& _bi);
     static Nonce nonce(BlockHeader const& _bi) { return _bi.seal<Nonce>(NonceField); }
     static h256 mixHash(BlockHeader const& _bi) { return _bi.seal<h256>(MixHashField); }
-    static h256 boundary(BlockHeader const& _bi) { auto d = _bi.difficulty(); return d ? (h256)u256((bigint(1) << 256) / d) : h256(); }
+
+    static h256 boundary(BlockHeader const& _bi)
+    {
+        const auto& d = _bi.difficulty();
+        return d > 1 ? h256{u256((u512(1) << 256) / d)} : ~h256{};
```

Comment:
It returned 0 before for d == 0, now it returns `ff...ff`, is this correct?

Reply:
Maybe make a `constexpr` constant out of `~h256{}`

Reply:
> Maybe make a `constexpr` constant out of `~h256{}`

How? We don't have any `constexpr` support in `h256`.

Reply:
Could be just const then.

Reply:
You mean `const h256 maxBoundary = ~h256{}`?

Reply:
Yes. Then it's guaranteed not to calculate negation each time.